dvc/remote/local.py

```diff
@@ -136,23 +128,25 @@ def remove(self, path_info):
         if self.exists(path_info):
             remove(path_info.fspath)
 
-    def move(self, from_info, to_info):
+    def move(self, from_info, to_info, mode=None):
         if from_info.scheme != "local" or to_info.scheme != "local":
             raise NotImplementedError
 
         self.makedirs(to_info.parent)
 
-        if self.isfile(from_info):
-            mode = self._file_mode
-        else:
-            mode = self._dir_mode
+        if mode is None:
+            if self.isfile(from_info):
+                mode = self._file_mode
+            else:
+                mode = self._dir_mode
```

Comment:
Not the nicest thing, but it was there from a "shared" PR, need to check if I can possibly make it more pretty by passing mode explicitly where `move` is used.